### PR TITLE
Improve ILLink.targets integration

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Runtime side of https://github.com/dotnet/sdk/pull/23470 (depends on that getting merged). Skip touching the IL Linker semaphore that tricks the target into not running.

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 1]
```diff
@@ -39,7 +40,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
     <IlcGenerateStackTraceData Condition="$(IlcGenerateStackTraceData) == ''">true</IlcGenerateStackTraceData>
     <IlcScanReflection Condition="$(IlcScanReflection) == ''">true</IlcScanReflection>
-    <SuppressTrimAnalysisWarnings Condition="$(SuppressTrimAnalysisWarnings) == ''">true</SuppressTrimAnalysisWarnings>
+    <SuppressTrimAnalysisWarnings Condition="$(SuppressTrimAnalysisWarnings) == '' and $([MSBuild]::VersionLessThan('$(NETCoreSdkVersion)', '6.0.0'))">true</SuppressTrimAnalysisWarnings>
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:43]: This is unrelated. We're only supposed to suppress the warnings in .NET 5. .NET 6 has the warnings enabled by default.

Comment (jkotas) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:43]: We should be able to remove the .NET 5 SDK support for the SDK. The team that we have agreed to support moved to .NET 6 SDK.

**Review by jkotas — APPROVED** [review]
